src/java.desktop/share/classes/javax/swing/TablePrintable.java

```diff
@@ -395,11 +395,14 @@ public int print(Graphics graphics, PageFormat pageFormat, int pageIndex)
         // draw a box around the table
         g2d.setColor(Color.BLACK);
 
+        Rectangle bounds = table.getBounds();
+        bounds.x = bounds.y = 0;
+
         // compute the visible portion of table and draw the rect around it
-        Rectangle visibleBounds = clip.intersection(table.getBounds());
+        Rectangle visibleBounds = clip.intersection(bounds);
         Point upperLeft = visibleBounds.getLocation();
-        Point lowerRight = new Point(visibleBounds.x + visibleBounds.width,
-                                     visibleBounds.y + visibleBounds.height);
+        Point lowerRight = new Point(visibleBounds.x + visibleBounds.width - 1,
+                                     visibleBounds.y + visibleBounds.height - 1);
 
```

Comment:
What is the effect of the -1 ? Meaning what happens if you don't do it - how did it contribute to the bug ?
And if this is about height why the -1 on the width too ?

Reply:
If not -1, then lowerRight would have been maxRow, as I Mentioned earlier minRow and maxRow in BasicTableUI is 0 and max-1, whereas in TablePrintable it was 1 to max. In order to match both computation for further calculation I made it similar to BasicTableUI class.